### PR TITLE
Support Mongo/DocDB stream Delete and other Operation types

### PR DESCRIPTION
**Pull request by dinujoh**
### Description
Support Mongo/DocDB stream Delete and other Operation types
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java [hunk 1]
```diff
@@ -175,35 +176,55 @@ public void processStream(final StreamPartition streamPartition) {
...
+                                    writeToBuffer(records, checkPointToken, recordCount);
+                                    lastLocalCheckpoint = checkPointToken;
+                                    lastLocalRecordCount = recordCount;
+                                    lastBufferWriteTime = System.currentTimeMillis();
+                                    records.clear();
+                                }
+                            } else if(shouldTerminateChangeStream(operationType)){
+                                stop();
+                                partitionCheckpoint.resetCheckpoint();
+                                LOG.warn("The change stream is invalid due to stream operation type {}. Stopping the change stream.", operationType);
```
Comment (graytaylor0) [inline data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java:213]: Will this message be misleading? Will users know that we will continue reading from the change stream?

**Review by graytaylor0 — COMMENTED** [review]
Changes look fine but build is failing

```
 /home/runner/work/data-prepper/data-prepper/data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java:212: error: cannot find symbol
                                partitionCheckpoint.resetCheckpoint();
```

**Comment by dinujoh** [comment]
> Changes look fine but build is failing
> 
> ```
>  /home/runner/work/data-prepper/data-prepper/data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java:212: error: cannot find symbol
>                                 partitionCheckpoint.resetCheckpoint();
> ```

File: data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java [hunk 2]
```diff
@@ -227,6 +248,19 @@ public void processStream(final StreamPartition streamPartition) {
         }
     }
 
+    private boolean isCRUDOperation(final OperationType operationType) {
+        return OperationType.INSERT == operationType ||
```
Comment (chenqi0805) [inline data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java:252]: MNT: suggest maintain a set of CRUD operation and check whether it is in the set

File: data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java [hunk 3]
```diff
@@ -227,6 +248,19 @@ public void processStream(final StreamPartition streamPartition) {
...
+                OperationType.DELETE == operationType ||
+                OperationType.UPDATE == operationType ||
+                OperationType.REPLACE == operationType;
+    }
+
+    private boolean shouldTerminateChangeStream(final OperationType operationType) {
+        return OperationType.INVALIDATE == operationType ||
```
Comment (chenqi0805) [inline data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java:259]: Similar as above

**Review by chenqi0805 — APPROVED** [review]

File: data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java (hunk 2 above)
Comment (dinujoh) [reply data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java]: Done

File: data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java (hunk 3 above)
Comment (dinujoh) [reply data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java]: Done

File: data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java (hunk 1 above)
Comment (dinujoh) [reply data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java]: Update the log message

**Review by graytaylor0 — APPROVED** [review]

**Pull request merged**